mmcv/runner/hooks/logger/pavi.py

```diff
@@ -100,6 +100,10 @@ def after_run(self, runner):
                     snapshot_file_path=ckpt_path,
                     iteration=iteration)
 
+        # flush the buffer and send a task ending signal to Pavi
+        if hasattr(self, 'writer'):
```

Comment:
What case the `self` object miss the attribute `writer`

Reply:
This part was placed in `__del__` so we checked the `writer` in case that `before_run` has not run and built the writer.
Since it's in `after_run` now which is always called after `before_run`, maybe this check is no longer necessary?

Reply:
yet